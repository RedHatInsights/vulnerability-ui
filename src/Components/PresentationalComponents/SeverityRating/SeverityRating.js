import { Text, TextVariants } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React from 'react';
import './SeverityRating.scss';

class SeverityRating extends React.Component {
    constructor(props) {
        super(props);

        this.severityTextList = {
            critical: 'This rating is given to flaws that could be easily exploited by a remote unauthenticated attacker'
                + ' and lead to system compromise (arbitrary code execution) without requiring user interaction.'
                + ' These are the types of vulnerabilities that can be exploited by worms. Flaws that require an authenticated'
                + ' remote user, a local user, or an unlikely configuration are not classed as Critical impact.',
            important: 'This rating is given to flaws that can easily compromise the confidentiality, integrity, or availability'
                + ' of resources. These are the types of vulnerabilities that allow local users to gain privileges, allow'
                + ' unauthenticated remote users to view resources that should otherwise be protected by authentication,'
                + ' allow authenticated remote users to execute arbitrary code, or allow remote users to cause a denial'
                + ' of service.',
            moderate: 'This rating is given to flaws that may be more difficult to exploit but could still lead to some'
                + ' compromise of the confidentiality, integrity, or availability of resources, under certain circumstances.'
                + ' These are the types of vulnerabilities that could have had a Critical impact or Important impact'
                + ' but are less easily exploited based on a technical evaluation of the flaw, or affect unlikely'
                + ' configurations.',
            low: 'This rating is given to all other issues that have a security impact. These are the types of vulnerabilities'
                + ' that are believed to require unlikely circumstances to be able to be exploited, or where a successful exploit'
                + ' would give minimal consequences.'
        };

        this.colorList = {
            default: 'severity-low',
            danger: 'severity-critical',
            warning: 'severity-moderate',
            orange: 'severity-color-important'
        };

        /* I'm not shure that the list of impacts is ordered correctly by relevancy */
        this.impactList = {
            Critical: {
                title: 'Critical',
                color: this.colorList.danger,
                text: this.severityTextList.critical
            },
            High: {
                title: 'High',
                color: this.colorList.orange,
                text: this.severityTextList.important
            },
            Important: {
                title: 'Important',
                color: this.colorList.orange,
                text: this.severityTextList.important
            },
            Medium: {
                title: 'Medium',
                color: this.colorList.warning,
                text: this.severityTextList.moderate
            },
            Moderate: {
                title: 'Moderate',
                color: this.colorList.warning,
                text: this.severityTextList.moderate
            },
            Low: {
                title: 'Low',
                color: this.colorList.default,
                text: this.severityTextList.low
            }
        };
    }

    getColoredTextByImpact(impact) {
        let text;

        if (Object.keys(this.impactList).includes(impact)) {
            text = {
                title: this.impactList[impact].title,
                color: this.impactList[impact].color,
                text: this.impactList[impact].text
            };
        } else {
            text = {
                title: 'Unknown',
                color: this.colorList.default,
                text: ''
            };
        }

        return text;
    }

    render() {
        const content = this.getColoredTextByImpact(this.props.impact);

        return (
            <React.Fragment>
                {this.props.format === 'title' ? (
                    <Text component={TextVariants.p} className={content.color}>{content.title}</Text>
                ) : (
                    <Text component={TextVariants.p}>{content.text}</Text>
                )}
            </React.Fragment>
        );
    }
}

SeverityRating.propTypes = {
    impact: propTypes.string,
    format: propTypes.string // (title) || text
};

export default SeverityRating;
