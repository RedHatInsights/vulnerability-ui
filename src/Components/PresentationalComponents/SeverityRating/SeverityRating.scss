@import "../../../App";

.severity-low {
    color: var(--pf-global--Color--200); // grey
}

.severity-critical {
    color: var(--pf-global--danger-color--100); // red
}

.severity-moderate {
    color: var(--pf-global--warning-color--100); // yellow
}

.severity-color-important {
    color: #ec7a08; // orange
}
