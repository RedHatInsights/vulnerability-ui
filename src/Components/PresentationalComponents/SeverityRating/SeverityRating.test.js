import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SeverityRating from './SeverityRating';

describe('SeverityRating', () => {
    it.each`
        impact_param
        ${'Critical'}
        ${'High'}
        ${'Important'}
        ${'Medium'}
        ${'Moderate'}
        ${'Low'}
        ${'Unknown'}
        ${undefined}
    `('render severity title', ({ impact_param }) => {
    const wrapper = shallow(<SeverityRating impact={impact_param} format={'title'} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it.each`
        impact_param
        ${'Critical'}
        ${'High'}
        ${'Important'}
        ${'Medium'}
        ${'Moderate'}
        ${'Low'}
        ${'Unknown'}
        ${undefined}
    `('render severity link', ({ impact_param }) => {
        const wrapper = shallow(<SeverityRating impact={impact_param} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
