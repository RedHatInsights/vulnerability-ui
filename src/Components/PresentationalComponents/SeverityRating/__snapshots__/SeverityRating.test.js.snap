// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeverityRating render severity link 1`] = `
<Fragment>
  <Text
    className=""
    component="p"
  >
    <a
      href="https://access.redhat.com/security/updates/classification"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more 
      <ExternalLinkAltIcon
        color="currentColor"
        size="sm"
        title={null}
      />
    </a>
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity link 2`] = `
<Fragment>
  <Text
    className=""
    component="p"
  >
    <a
      href="https://access.redhat.com/security/updates/classification"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more 
      <ExternalLinkAltIcon
        color="currentColor"
        size="sm"
        title={null}
      />
    </a>
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity link 3`] = `
<Fragment>
  <Text
    className=""
    component="p"
  >
    <a
      href="https://access.redhat.com/security/updates/classification"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more 
      <ExternalLinkAltIcon
        color="currentColor"
        size="sm"
        title={null}
      />
    </a>
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity link 4`] = `
<Fragment>
  <Text
    className=""
    component="p"
  >
    <a
      href="https://access.redhat.com/security/updates/classification"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more 
      <ExternalLinkAltIcon
        color="currentColor"
        size="sm"
        title={null}
      />
    </a>
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity link 5`] = `
<Fragment>
  <Text
    className=""
    component="p"
  >
    <a
      href="https://access.redhat.com/security/updates/classification"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more 
      <ExternalLinkAltIcon
        color="currentColor"
        size="sm"
        title={null}
      />
    </a>
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity link 6`] = `
<Fragment>
  <Text
    className=""
    component="p"
  >
    <a
      href="https://access.redhat.com/security/updates/classification"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more 
      <ExternalLinkAltIcon
        color="currentColor"
        size="sm"
        title={null}
      />
    </a>
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity link 7`] = `
<Fragment>
  <Text
    className=""
    component="p"
  >
    <a
      href="https://access.redhat.com/security/updates/classification"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more 
      <ExternalLinkAltIcon
        color="currentColor"
        size="sm"
        title={null}
      />
    </a>
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity link 8`] = `
<Fragment>
  <Text
    className=""
    component="p"
  >
    <a
      href="https://access.redhat.com/security/updates/classification"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more 
      <ExternalLinkAltIcon
        color="currentColor"
        size="sm"
        title={null}
      />
    </a>
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity title 1`] = `
<Fragment>
  <Text
    className="severity-critical cve-severity-title"
    component="p"
  >
    Critical
     impact
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity title 2`] = `
<Fragment>
  <Text
    className="severity-color-important cve-severity-title"
    component="p"
  >
    High
     impact
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity title 3`] = `
<Fragment>
  <Text
    className="severity-color-important cve-severity-title"
    component="p"
  >
    Important
     impact
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity title 4`] = `
<Fragment>
  <Text
    className="severity-moderate cve-severity-title"
    component="p"
  >
    Medium
     impact
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity title 5`] = `
<Fragment>
  <Text
    className="severity-moderate cve-severity-title"
    component="p"
  >
    Moderate
     impact
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity title 6`] = `
<Fragment>
  <Text
    className="severity-low cve-severity-title"
    component="p"
  >
    Low
     impact
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity title 7`] = `
<Fragment>
  <Text
    className="severity-low cve-severity-title"
    component="p"
  >
    Unknown
     impact
  </Text>
</Fragment>
`;

exports[`SeverityRating render severity title 8`] = `
<Fragment>
  <Text
    className="severity-low cve-severity-title"
    component="p"
  >
    Unknown
     impact
  </Text>
</Fragment>
`;
